components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -2130,6 +2145,14 @@ subroutine ice_rain_collection(rho,t,rhofaci,    &
    real(rtype), intent(out) :: qrcol
    real(rtype), intent(out) :: nrcol
 
+#ifdef SCREAM_CONFIG_IS_CMAKE
+    if (use_cxx) then
+       call ice_rain_collection_f(rho, t, rhofaci, logn0r, f1pr07, f1pr08, &
+                                  qitot_incld, nitot_incld, qr_incld, qrcol,nrcol)
+       return
+    endif
+#endif
+
    if (qitot_incld.ge.qsmall .and. qr_incld.ge.qsmall) then
       if (t.le.zerodegc) then
          ! note: f1pr08 and logn0r are already calculated as log_10
```

Comment:
@xyuan , I see the reason your results are not BFB. In the code below, you need to change the lines using the `**` operator to use bfb_pow.

Reply:
As an example:

This line
```f90
qrcol = 10._rtype**(f1pr08+logn0r)*rho*rhofaci*eri*nitot_incld
```

becomes
```f90
qrcol = bfb_pow(10._rtype, (f1pr08+logn0r)*rho*rhofaci*eri*nitot_incld)
```

In order for bfb_pow to work, you'll need to import cxx_pow from micro_p3_iso_f.